webhook.go

```diff
@@ -222,218 +228,236 @@ func hookHandler(w http.ResponseWriter, r *http.Request) {
 
 	log.Printf("[%s] incoming HTTP request from %s\n", rid, r.RemoteAddr)
 
+	id := mux.Vars(r)["id"]
+
+	matchedHook := matchLoadedHook(id)
+	if matchedHook == nil {
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Fprint(w, "Hook not found.")
+		return
+	}
+
+	// Check for allowed methods
+	if len(matchedHook.HTTPMethods) != 0 {
+		var allowed bool
+		for i := range matchedHook.HTTPMethods {
+			if matchedHook.HTTPMethods[i] == r.Method {
+				allowed = true
+				break
+			}
+		}
+
+		if !allowed {
+			w.WriteHeader(http.StatusMethodNotAllowed)
```

Comment:
It would be nice if we added log message so that user is aware when these situations happen.

Reply:
I.e. disallowed method used for hook xyz.

Reply:
Done.